### PR TITLE
Add book to system #17

### LINKED ISSUE #17: User Story 12, Add a New Book to the System
```
As a Visitor,
When I visit the book index page,
I see a link that allows me to add a new book.
When I click that link, I am taken to a new book path.
I can fill in a form to add a new book.
The form includes the following items:
- title
- year book was published
- number of pages (0 or higher)
- image of the book cover (optional, can be left blank)
- a list of authors (see notes below)

When I submit the form, I am taken to that book's show page.

Book titles should be converted to Title Case before saving.
Book titles should be unique within the system.
For authors, a comma-separated list of names should be entered,
and each author will be added to the database.
Authors added to the database should have their names converted
to Title Case.
Author Names should be unique within the system.
If image field is left blank, a default image should be set.
```

### Checklist:

- [ ] base code tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written
- [ ] edge case code written
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #18

child of #33

depends on #20

### PR DESCRIPTION
**Pull request by Vjp888**
Adds new and create actions to the books controller.

Create action will be able to separate authors and create or find authors to attach to the book.

Adds numericality checks to models and tests them.

Successfully redirects to the book's show page when book in made
will tell user what errors have occurred when a book is not made and fill out fields with previous info.

closes #17

### CODE REVIEW

File: spec/models/book_spec.rb [hunk 1]
```diff
@@ -6,6 +6,9 @@
     it { should validate_presence_of :title }
     it { should validate_presence_of :pages }
     it { should validate_presence_of :year_published }
+
+    it { should validate_numericality_of(:pages).only_integer}
+    it {should validate_numericality_of(:year_published).only_integer}
```
Comment (milevy1) [inline spec/models/book_spec.rb]: Can you add spaces between the { brackets } to keep it looking consistent?

File: spec/models/review_spec.rb [hunk 2]
```diff
@@ -6,6 +6,8 @@
     it { should validate_presence_of :title }
     it { should validate_presence_of :description }
     it { should validate_presence_of :username }
+
+    it {should validate_numericality_of(:rating).only_integer}
```
Comment (milevy1) [inline spec/models/review_spec.rb]: Here too, spaces between { brackets }

File: app/views/books/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  </p>
+
+  <p>
+    <%= f.label :year_published %>
+    <%= f.text_field :year_published %>
+  </p>
+
+  <p>
+    <label for="author_list">Authors</label>
+    <input type="text" name="author_list" placeholder="Optional">
```
Comment (milevy1) [inline app/views/books/new.html.erb]: Author_list should not be optional, we need at least one for a new book

File: app/controllers/books_controller.rb [hunk 4]
```diff
@@ -6,4 +6,31 @@ def index
...
     @book = Book.find(params[:id])
   end
+
+  def new
+    @book = Book.new
+  end
+
+  def create
+    @book = Book.create(book_params)
+    # binding.pry
```
Comment (milevy1) [inline app/controllers/books_controller.rb]: Delete binding.pry
